### PR TITLE
Automated TCs for the observability-crd-status test case.

### PR DESCRIPTION
**Pull request by greyerof**
- Added apiextensions interface to the go-client.
- Created globalhelpers for CRD creation/deletion.
- Added support for targetCrdFilters for the TNF yaml config file.

### CODE REVIEW

File: tests/globalhelper/crd.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		}
+
+		for _, condition := range runningCrd.Status.Conditions {
+			if condition.Type == apiextv1.Established {
+				return true
+			}
+		}
+
+		return false
+	}, timeout, crdRetryIntervalSecs*time.Second).Should(Equal(true), "CRD is not ready")
```
Comment (theemj) [inline tests/globalhelper/crd.go:49]: Might make sense to move the *time.Second up to the const definition since it's used twice in the same way

**Review by theemj — APPROVED** [review]
just one style comment  that wouldn't change the code functionality either way. looks good to me!

File: tests/globalhelper/crd.go (hunk 1 above)
Comment (greyerof) [reply tests/globalhelper/crd.go]: Done.

File: tests/globalhelper/crd.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package globalhelper
```
Comment (kononovn) [inline tests/globalhelper/crd.go:1]: Function from this module is used only in observability suite. So better to keep them in observability/helper

**Review by kononovn — CHANGES_REQUESTED** [review]
Looks good except 1 comment above.

File: tests/globalhelper/crd.go (hunk 2 above)
Comment (greyerof) [reply tests/globalhelper/crd.go]: Done!

**Review by kononovn — APPROVED** [review]
lgtm

**Pull request merged**